deepchem/feat/graph_features.py

```diff
@@ -263,14 +290,30 @@ class WeaveFeaturizer(Featurizer):
 
   name = ['weave_mol']
 
-  def __init__(self):
+  def __init__(self, graph_distance=True, explicit_H=None):
+    # Distance is either graph distance(True) or Euclidean distance(False,
+    # only support datasets providing Cartesian coordinates)
+    self.graph_distance = graph_distance
     # Set dtype
     self.dtype = object
+    # Check if there are explicit hydrogens, default to be False
+    self.check_H = False
+    if explicit_H is None:
+      self.explicit_H = False
+      # Set to True if explicit hydrogen is not specified
+      self.check_H = True
 
   def _featurize(self, mol):
     """Encodes mol as a WeaveMol object."""
+    # Check hydrogen in the molecule
+    if self.check_H and not self.explicit_H:
+      for a in mol.GetAtoms():
+        if a.GetSymbol() == 'H':
+          self.explicit_H = True
```

Comment:
I actually don't know, since rarely organic molecules do not have hydrogens, maybe we could just get rid of this check hydrogen part and specify with or without hydrogen upon calling.